src/foundry/foundry.js/canvas.d.ts

```diff
@@ -17,401 +20,498 @@
+     * @defaultValue `null`
+     */
+    scene: InstanceType<ConfiguredDocumentClass<typeof Scene>> | null;
+
+    /**
+     * The current pixel dimensions of the displayed Scene, or null if the Canvas is blank.
+     * @defaultValue `null`
+     */
+    dimensions: Canvas.Dimensions | null;
+
+    /**
+     * A reference to the HeadsUpDisplay container which overlays HTML rendering on top of this Canvas.
+     * @defaultValue `null`
+     */
+    hud: HeadsUpDisplay | null;
+
+    /**
+     * An Array of pending canvas operations which should trigger on the next re-paint
+     * @defaultValue `[]`
+     */
+    pendingOperations: Array<[(args: any[]) => void, any, any[]]>;
+
+    /**
+     * A perception manager interface for batching lighting, sight, and sound updates
+     */
+    perception: PerceptionManager;
+
+    /**
+     * A flag for whether the game Canvas is ready to be used. False if the canvas is not yet drawn, true otherwise.
+     * @defaultValue `false`
+     */
+    ready: boolean;
+
+    /**
+     * A flag for whether the game Canvas is initialized and ready for drawing.
+     * @defaultValue `false`
+     */
+    initialized: boolean;
+
+    /**
+     * Track the timestamp of the last stage zoom operation
+     * @defaultValue `0`
+     */
+    protected _zoomTime: number;
+
+    /**
+     * Track the last automatic pan time to throttle
+     * @defaultValue `0`
+     */
+    protected _panTime: number;
 
-  notes: NotesLayer | undefined;
+    /**
+     * An object of data which is temporarily cached to be reloaded after the canvas is drawn
+     * @defaultValue `{}`
+     */
+    protected _reload: Partial<{ scene: string; layer: string; controlledTokens: string[]; targetedTokens: string[] }>;
 
-  tokens: TokenLayer | undefined;
-
-  lighting: LightingLayer | undefined;
-
-  sounds: SoundsLayer | undefined;
-
-  sight: SightLayer | undefined;
-
-  effects: EffectsLayer | undefined;
-
-  controls: ControlsLayer | undefined;
+    /**
+     * A Set of unique pending operation names to ensure operations are only performed once
+     */
+    protected _pendingOperationNames: Set<string>;
+
+    /**
+     * The pixel radius of blur distance that should be applied for the current zoom level
+     * @defaultValue `0`
+     */
+    blurDistance: number;
+
+    /**
+     * An array of blur filter instances which are modified by the zoom level and the "soft shadows" setting
+     * @defaultValue `[]`
+     */
+    blurFilters: PIXI.filters.BlurFilter[];
+
+    /**
+     * The singleton interaction manager instance which handles mouse interaction on the Canvas.
+     */
+    mouseInteractionManager: MouseInteractionManager<Exclude<this['stage'], undefined>> | undefined;
+
+    /**
+     * A reference to the MouseInteractionManager that is currently controlling pointer-based interaction, or null.
+     * @defaultValue `null`
+     */
+    currentMouseManager: MouseInteractionManager<Exclude<this['stage'], undefined>> | null;
```

Comment:
Why reference via `this['stage']`? Just using `PIXI.Container` might be simpler and should be good enough...?